src/corelibs/U2View/src/ov_sequence/GSequenceGraphView.cpp

```diff
@@ -247,69 +232,49 @@ void GSequenceGraphView::sl_onShowVisualProperties(bool) {
 
 void GSequenceGraphView::sl_onDeleteAllLabels() {
     foreach (const QSharedPointer<GSequenceGraphData> graph, graphs) {
-        graph->graphLabels.deleteAllLabels();
+        graph->labels.deleteAllLabels();
     }
 }
 
-void GSequenceGraphView::sl_onSelectExtremumPoints() {
-    const QRect &graphRect = static_cast<GSequenceGraphViewRA *>(renderArea)->getGraphRect();
-    QObjectScopedPointer<GraphLabelsSelectDialog> dlg = new GraphLabelsSelectDialog(getSequenceLength(), this);
-    dlg->exec();
-    CHECK(!dlg.isNull(), );
-
-    if (dlg->result() == QDialog::Accepted) {
-        int windowSize = dlg->getWindowSize();
-        bool usingIntervals = dlg->isUsedIntervals();
-        const QVector<U2Region> &selection = getSequenceContext()->getSequenceSelection()->getSelectedRegions();
-        foreach (const QSharedPointer<GSequenceGraphData> graph, graphs) {
-            if (true == usingIntervals) {
-                foreach (const U2Region &selectedRegion, selection) {
-                    graphDrawer->selectExtremumPoints(graph, graphRect, windowSize, selectedRegion);
-                }
-            } else
-                graphDrawer->selectExtremumPoints(graph, graphRect, windowSize, getVisibleRange());
+void GSequenceGraphView::sl_showLocalMinMaxLabels() {
+    QVector<U2Region> regionsToAnnotate = getSequenceContext()->getSequenceSelection()->getSelectedRegions();
+    if (regionsToAnnotate.isEmpty()) {
+        regionsToAnnotate.append(getVisibleRange());
+    }
+    const QRect &graphRect = getGraphRenderArea()->getGraphRect();
+    for (QSharedPointer<GSequenceGraphData> &graph : graphs) {
```

Comment:
`for (const QSharedPointer<GSequenceGraphData> &graph : graphs) {`

Reply:
This one is good. Fixed